core/src/main/java/org/elasticsearch/index/query/WrapperQueryBuilder.java

```diff
@@ -134,4 +127,21 @@ protected int doHashCode() {
     protected boolean doEquals(WrapperQueryBuilder other) {
         return Arrays.equals(source, other.source);   // otherwise we compare pointers
     }
+
+    @Override
+    public QueryBuilder<?> rewrite(QueryRewriteContext context) throws IOException {
+        try (XContentParser qSourceParser = XContentFactory.xContent(source).createParser(source)) {
+            QueryParseContext parseContext = context.newParseContext();
+            parseContext.reset(qSourceParser);
+
+            final QueryBuilder<?> queryBuilder = parseContext.parseInnerQueryBuilder();
+            if (queryBuilder.boost() == DEFAULT_BOOST) {
```

Comment:
I think this is not correct, as we will ignore one of the boosts. previously the two boosts 9if they both had a value != 1.0f) caused the queries to be wrapped twice with BoostQuery, hence the two boosts ended up being multiplied by lucene. I think we may want to multiply ourselves here and in WrapperQueryBuilder.